src/Agent.Listener/DistributedTask.Pipelines/Yaml/TypeConverters/ConverterUtil.processes.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.Contracts;
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.TypeConverters
+{
+    internal static partial class ConverterUtil
+    {
+        internal static IList<ProcessResource> ReadProcessResources(IParser parser)
+        {
+            var result = new List<ProcessResource>();
+            parser.Expect<SequenceStart>();
+            while (parser.Allow<SequenceEnd>() == null)
+            {
+                parser.Expect<MappingStart>();
+                Scalar scalar = parser.Expect<Scalar>();
+                switch (scalar.Value ?? String.Empty)
+                {
+                    case YamlConstants.Endpoint:
+                    case YamlConstants.Repo:
+                        break;
+                    default:
+                        throw new SyntaxErrorException(scalar.Start, scalar.End, $"Unexpected resource type: '{scalar.Value}'");
+                }
+
+                var resource = new ProcessResource { Type = scalar.Value };
```

Comment:
The form is `- type: resourceName`, then everything following property gets put into the Data dictionary.